### PR TITLE
feat: As a user, I want to differentiate the Attestations from each network

### LINKED ISSUE #426: [US] As a user, I want to differentiate the Attestations from each network
As a user,
I want to differentiate the Attestations from each network,
So that I get Attestations from multiple networks

### PR DESCRIPTION
**Pull request by alainncls**
## What does this PR do?

Adds a "chain prefix" to differentiate the Attestations from multiple networks

### Related ticket

Fixes #426 

### Type of change

- [ ] Chore
- [ ] Bug fix
- [X] New feature
- [ ] Documentation update

## Check list

- [X] Unit tests for any smart contract change
- [X] Contracts and functions are documented

### CODE REVIEW

**Review by orbmis — APPROVED** [review]

File: contracts/script/deploy/deployEverything.ts [hunk 1]
```diff
@@ -135,6 +136,13 @@ async function main() {
   await attestationRegistry.updateRouter(routerProxyAddress);
   console.log("AttestationRegistry updated!");
 
+  console.log("Updating AttestationRegistry with the chain prefix...");
+  const network = await ethers.provider.getNetwork();
```
[the remark this answers is not in the document]
Comment (alainncls) [reply contracts/script/deploy/deployEverything.ts:140]: Good point, let me change that!

[the remark this answers is not in the document]
Comment (alainncls) [reply contracts/script/deploy/deployEverything.ts:140]: I made the change there: https://github.com/Consensys/linea-attestation-registry/pull/427/files#diff-cb68705a5913d75772866cfde7825ef5ab4667f8bfc7931f4bb3142e294b282dR12

**Review by satyajeetkolhapure — APPROVED** [review]

File: contracts/src/AttestationRegistry.sol [hunk 2]
```diff
@@ -296,4 +316,23 @@ contract AttestationRegistry is OwnableUpgradeable {
     }
     return result;
   }
+
+  /**
+   * @notice Generate an attestation ID, prefixed by the Verax chain identifier
+   * @param id The attestation ID (coming after the chain prefix)
+   * @return The attestation ID
+   */
+  function generateAttestationId(uint256 id) public view returns (bytes32) {
```
Comment (0xEillo) [inline contracts/src/AttestationRegistry.sol:325]: @alainncls could this function be internal maybe ?

Comment (0xEillo) [reply contracts/src/AttestationRegistry.sol:325]: I feel like having the chain prefix as a bytes32 as oppose to uint256, then we can simply do:

bytes32(abi.encode(prefix+ counter))

Comment (alainncls) [reply contracts/src/AttestationRegistry.sol]: Done ✅ 
It's still tested and used in tests, via a "harness contract" (cf. [Foundry doc](https://book.getfoundry.sh/tutorials/best-practices#internal-functions)).

**Pull request merged**